### PR TITLE
Close streams properly

### PR DESCRIPTION
**Pull request by minborg**

### CODE REVIEW

File: src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java [hunk 1]
```diff
@@ -2476,7 +2481,8 @@ public void testTailerWhenCyclesWhereSkippedOnWrite() {
         }
     }
 
-    private void doSomething(@NotNull ThreadLocal<ExcerptAppender> tla, @NotNull ThreadLocal<ExcerptTailer> tlt, String text) {
+    private void doSomething
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java:2471]: can't see the benefit of auto-reformatting in this PR - it is occluding the real changes

File: src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java [hunk 2]
```diff
@@ -2146,7 +2151,7 @@ public void testLastIndexAppended() {
             appender.writeDocument(w -> w.writeEventName("hello").text("world0"));
             final long nextIndexToWrite = appender.lastIndexAppended() + 1;
             appender.writeDocument(w -> w.getValueOut().bytes(new byte[0]));
-                       // System.out.println(chronicle.dump());
+            // System.out.println(chronicle.dump());
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java:2142]: can't we just get rid of commented-out code?

**Review by nicktindall — APPROVED** [review]

**Review by JerryShea — APPROVED** [review]

File: src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java [hunk 3]
```diff
@@ -618,12 +619,13 @@ public void shouldAllowDirectoryToBeDeletedWhenQueueIsClosed() throws IOExceptio
             }
         }
 
-        Files.walk(dir.toPath()).forEach(p -> {
-            if (!Files.isDirectory(p)) {
-                assertTrue(p.toString(), p.toFile().delete());
-            }
-        });
-
+        try (Stream<Path> paths = Files.walk(dir.toPath())) {
```
Comment (JerryShea) [inline src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java:622]: can't use net.openhft.chronicle.core.io.IOTools#deleteDirWithFilesOrXXX ?

File: src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java (hunk 1 above)
Comment (minborg) [reply src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java:2471]: It accidentally happened. Sorry.

File: src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java (hunk 3 above)
Comment (minborg) [reply src/test/java/net/openhft/chronicle/queue/impl/single/SingleChronicleQueueTest.java:622]: That will delete the files but will not assert "deletability".

**Pull request merged**